### PR TITLE
Remove snap capability when SnapSync is finished

### PR DESCRIPTION
**Pull request by marcindsobczak**

### CODE REVIEW

File: src/Nethermind/Nethermind.Network/ProtocolsManager.cs [hunk 1]
```diff
@@ -412,6 +412,15 @@ public void AddSupportedCapability(Capability capability)
 
             _capabilities.Add(capability);
         }
+        
+        public void RemoveSupportedCapability(Capability capability)
+        {
+            if (_capabilities.Contains(capability))
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network/ProtocolsManager.cs:418]: Maybe _capabilities should be HashSet?
Will that break order? Is order of capabilities needed?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Network/SnapCapabilitySwitcher.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+{
+    private readonly IProtocolsManager _protocolsManager;
+    private readonly ProgressTracker _progressTracker;
+
+    public SnapCapabilitySwitcher(IProtocolsManager? protocolsManager, ProgressTracker? progressTracker)
+    {
+        _protocolsManager = protocolsManager ?? throw new ArgumentNullException(nameof(protocolsManager));
+        _progressTracker = progressTracker ?? throw new ArgumentNullException(nameof(progressTracker));
+
+        _progressTracker.SnapSyncFinished += OnSnapSyncFinished;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network/SnapCapabilitySwitcher.cs:37]: btw can we check and attach only when snap sync will be used? So if we are already in sync don't attach? Or attach to some check (multi sync mode selector?) and detach here when snap sync won't be used?

File: src/Nethermind/Nethermind.Network/ProtocolsManager.cs [hunk 3]
```diff
@@ -405,12 +405,15 @@ private void AddNodeToDiscovery(ISession session, P2PProtocolInitializedEventArg
 
         public void AddSupportedCapability(Capability capability)
         {
-            if (_capabilities.Contains(capability))
+            _capabilities.Add(capability);
```
Comment (dceleda) [inline src/Nethermind/Nethermind.Network/ProtocolsManager.cs:408]: Why we don't check 
`if (_capabilities.Contains(capability))`
anymore ?

**Review by dceleda — APPROVED** [review]

File: src/Nethermind/Nethermind.Network/ProtocolsManager.cs (hunk 3 above)
Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Network/ProtocolsManager.cs:408]: No need after changing from `List<Capability>` to `HashSet<Capability>`. `Add(item)` for `HashSet` is `AddIfNotPresent(item)` actually. https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1.add?view=net-6.0

Comment (dceleda) [reply src/Nethermind/Nethermind.Network/ProtocolsManager.cs:408]: Cool. Missed the change.

**Pull request merged**